cli/tests/unit/http_test.ts

```diff
@@ -1165,14 +1165,513 @@ Deno.test(
 
     const resp = new Uint8Array(200);
     const readResult = await conn.read(resp);
-    assertEquals(readResult, 115);
+    assertEquals(readResult, 138);
 
     conn.close();
 
     await promise;
   },
 );
 
+/* Automatic Body Compression */
+
+const decoder = new TextDecoder();
+
+Deno.test({
+  name: "http server compresses body",
+  permissions: { net: true },
+  async fn() {
+    const hostname = "localhost";
+    const port = 4501;
+
+    async function server() {
+      const listener = Deno.listen({ hostname, port });
+      const tcpConn = await listener.accept();
+      const httpConn = Deno.serveHttp(tcpConn);
+      const e = await httpConn.nextRequest();
+      assert(e);
+      const { request, respondWith } = e;
+      assertEquals(request.headers.get("Accept-Encoding"), "gzip, deflate, br");
+      const response = new Response(
+        JSON.stringify({ hello: "deno", now: "with", compressed: "body" }),
+        {
+          headers: { "content-type": "application/json" },
+        },
+      );
+      await respondWith(response);
+      httpConn.close();
+      listener.close();
+    }
+
+    async function client() {
+      const url = `http://${hostname}:${port}/`;
+      const cmd = [
+        "curl",
```

Comment:
nit: should use `fetch()`?

Reply:
No, because `fetch()` as per the spec doesn't allow control of the `Accept-Encoding` header, as it is one of the forbidden headers and Deno sets is specifically, and automatically decodes the body before sending it back in the `Response()`.  We need to a more direct method for testing purposes.